packages/graphql-language-service-server/src/GraphQLLanguageService.ts

```diff
@@ -96,11 +100,38 @@ export class GraphQLLanguageService {
     this._logger = logger;
   }
 
-  getConfigForURI(uri: Uri) {
-    const config = this._graphQLCache.getProjectForFile(uri);
-    if (config) {
-      return config;
+  getAllProjectsForFile(uri: Uri) {
+    const filePath = uri.startsWith('file:') ? fileURLToPath(uri) : uri;
+    const projects = Object.values(this._graphQLConfig.projects).filter(
+      project => project.match(filePath),
+    );
+
+    return projects.length > 0
+      ? projects
+      : // Fallback, this always finds at least 1 project
+        [this._graphQLConfig.getProjectForFile(filePath)];
+  }
+
+  getProjectForQuery(
+    query: string,
+    uri: Uri,
+    projects?: GraphQLProjectConfig[],
+  ) {
+    if (!query.startsWith('#graphql')) {
+      // Query is not annotated with #graphql.
+      // Skip suffix check and return the first project that matches the file.
+      return projects?.[0] ?? this._graphQLConfig.getProjectForFile(uri);
     }
```

Comment:
Note: I didn't want to modify the existing behavior of other patterns so I didn't fix the `file://` protocol bug here. Since it's acknowledged that it's a bug we should fix, this line should be updated to remove `file://` as well.

Reply:
It now removes `file://` here as well to prevent the bug.